trie/utils/nodes.py

```diff
@@ -89,3 +101,62 @@ def consume_common_prefix(left_key, right_key):
 
 def key_starts_with(full_key, partial_key):
     return all(left == right for left, right in zip(full_key, partial_key))
+
+
+# Binary Trie node utils
+def parse_node(node):
+    """
+    Input: a serialized node
+    """
+    if node is None or node == b'':
+        raise InvalidNode("Blank node is not a valid node type in Binary Trie")
+    elif node[0] == BRANCH_TYPE:
+        if len(node) != 65:
+            raise InvalidNode("Invalid branch node, both child node should be 32 bytes long each")
+        # Output: left child, right child, node type
+        return node[1:33], node[33:], BRANCH_TYPE
+    elif node[0] == KV_TYPE:
+        if len(node) <= 33:
+            raise InvalidNode("Invalid kv node, short of key path or child node hash")
+        # Output: keypath: child, node type
+        return decode_to_bin_keypath(node[1:-32]), node[-32:], KV_TYPE
+    elif node[0] == LEAF_TYPE:
+        if len(node) == 1:
+            raise InvalidNode("Invalid leaf node, can not contain empty value")
+        # Output: None, value, node type
+        return None, node[1:], LEAF_TYPE
+    else:
+        raise InvalidNode("Unable to parse node")
+
+
+def encode_kv_node(keypath, child_node_hash):
+    """
+    Serializes a key/value node
+    """
+    if keypath is None:
+        raise ValidationError("Key path can not be empty")
+    validate_is_bytes(keypath)
+    validate_is_bytes(child_node_hash)
+    validate_length(child_node_hash, 32)
+    return bytes([KV_TYPE]) + encode_from_bin_keypath(keypath) + child_node_hash
```

Comment:
We can probably gain some *minor* performance improvements if we stored `KV_NODE_PREFIX = bytes([KV_TYPE])` as a constant.

Same for the other prefixes.